tekton/mario-bot/mario-github-comment.yaml

```diff
@@ -42,117 +42,154 @@ spec:
+    description: The fully qualified image target e.g. repo/name:tag
   - name: passedOrFailed
     description: Whether the triggering event was successful or not
   resourcetemplates:
   - apiVersion: tekton.dev/v1beta1
-    kind: TaskRun
+    kind: PipelineRun
     metadata:
       generateName: mario-comment-github-
     spec:
       serviceAccountName: mario-listener
-      taskSpec:
-        resources:
-          inputs:
-            - name: source
-              type: git
-            - name: pr
-              type: pullRequest
-          outputs:
-            - name: image
-              type: image
-            - name: pr
-              type: pullRequest
-        steps:
-        - name: copy-pr-to-output
-          image: busybox
-          script: |
-            #!/bin/sh
-            mkdir -p $(resources.outputs.pr.path)
-            cp -r $(resources.inputs.pr.path)/* $(resources.outputs.pr.path)/
-        - name: setup-comment
-          image: python:3-alpine
-          script: |
-            #!/usr/bin/env python
-            import json
-            import random
-
-            marios_pics_root = 'https://storage.googleapis.com/mario-bot/pics'
-            ok_pics = ['mario', 'luigi', 'tekton']
-            failed_pics = ['goomba']
-            logs_url = 'http://35.222.249.224/?buildid=%s&namespace=mario'
-            successful = ("$(tt.params.passedOrFailed)" == "True")
-            print("PassedOrFailed: {}".format("$(tt.params.passedOrFailed)"))
-
-            # Service Image
-            comment_template = (
-              '<img width="200" alt="{pic_alt}" src="{pic_src}">'
-              ' at your service! </p>'
+      pipelineRef:
+        name: mario-comment-github
+      workspaces:
+      - name: pr
+        volumeClaimTemplate:
+          spec:
+            accessModes:
+            - ReadWriteOnce
+            resources:
+              requests:
+                storage: 10Mi
+      params:
+      - name: pullRequestID
+        value: $(tt.params.pullRequestID)
+      - name: buildUUID
+        value: $(tt.params.buildUUID)
+      - name: gitURL
+        value: $(tt.params.gitURL)
+      - name: targetImage
+        value: $(tt.params.targetImage)
+      - name: passedOrFailed
+        value: $(tt.params.passedOrFailed)
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: mario-comment-github
+spec:
+  params:
+  - name: pullRequestID
+  - name: buildUUID
+  - name: gitURL
+  - name: targetImage
+  - name: passedOrFailed
+  workspaces:
+  - name: pr
+  tasks:
+  - name: download-pr-info
+    taskRef:
+      name: pull-request
+      bundle: gcr.io/tekton-releases/catalog/upstream/pull-request:0.1
+    workspaces:
+    - name: pr
+      workspace: pr
+    params:
+    - name: mode
+      value: download
+    - name: url
+      value: $(params.gitURL)/pull/$(params.pullRequestID)
+    - name: provider
+      value: github
+    - name: secret-key-ref
+      value: mario-github-token
```

Comment:
This seems fine - we also have a custom interceptor available on `dogfooding` that extends the event payload with the whole PR data - which is used for the same use case as here - i.e. that the payload of the comment event in GitHub only includes very little data on the PR itself (the URL mostly).

This is an example of how it is used:
https://github.com/tektoncd/plumbing/blob/def914a109d482355453d7f6dc766e476a371705/tekton/ci-workspace/plumbing/trigger.yaml#L48-L56

The main advantage of the interceptor is that is saves you the extra pod for the task, and it allows to always have the same input params, regardless of the github specific event.

I don't mind using the task, but you might consider switching to the interceptor in a follow-up PR

Reply:
Added a TODO to use the interceptor.